### ISSUE TITLE
Cart empty after the customer login

### ISSUE THREAD

**Issue by shinesoftware** [magento/magento2#12415]
# Preconditions
Ubuntu, Standard LAMP (Apache, MySQL, ....)
Magento ver. 2.1.9 - Clean setup by command line
PHP 7.0.8-0ubuntu0.16.04.3 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
mysql Ver 14.14 Distrib 5.7.17, for Linux (x86_64) using EditLine wrapper

# Installation setup 
    bin/magento' -- setup:install --language='it_IT' --timezone='Europe/Rome' --db-host='localhost' --db-name='magento_219' --db-user='root' --base-url='http://magento219.shinesoftware.it/' --use-rewrites='1' --use-secure='0' --use-secure-admin='1' --admin-user='admin' --admin-lastname='Doe' --admin-firstname='John' --admin-email='john.doe@example.com' --admin-password='magento2017' --session-save='files' --backend-frontname='admin' --currency='EUR' --db-password='[password]' --base-url-secure='https://magento219.shinesoftware.it/'

# Steps to reproduce

Add a product in your basket case
Go to your cart summary page and login as customer

# Expected result
Magento must login the user and the cart must not emptied!

# Actual result
Magento clear the cart after the customer login

# Video
https://youtu.be/689tv8Ldrww

thanks

**Comment by Sil-B** [comment]
Same problem. @shinesoftware can you check if setting "Redirect Customer to Account Dashboard after Logging in" to "no", solves the issue? Customer -> Customer Configuration

**Comment by magento-engcom-team** [comment]
Hello @shinesoftware. Thanks for reporting. Currently we are not able to reproduce Your issue. Would You please look if there are any exception logs or error reports after Login has been clicked.

**Comment by shinesoftware** [comment]
Hi guys, the problem was the domain name resolution (DNS). 

Here my steps:

- I have installed the project in the local server
- I have set the dns zone in the domain property in the hosting control panel. Magento base-url was a real DNS A Zone 

I have realized that some dns function, cookie features, magento or browser domain resolution do not recognize the domain zone and the cache was invalidated.

I have published the project in the local server with a new base-url magento219.local and fix the apache configuration and the hosts file and it works as well.

The problem has been solved!
